plugins/by-name/noice/default.nix

```
@@ -1,35 +1,103 @@
 {
   lib,
-  helpers,
-  config,
-  pkgs,
   ...
 }:
-# TODO: This uses a lot of types.anything because noice.nvim types are quite complex.
-# It should be possible to map them to nix, but they would not map really well through
-# toLuaObject, we would maybe need some ad-hoc pre-processing functions.
-with lib;
-{
-  options.plugins.noice = helpers.neovim-plugin.extraOptionsOptions // {
-    enable = mkEnableOption ''
-      noice.nvim, an experimental nvim UI.
-      Note that if treesitter is enabled you need the following parsers:
-        vim, regex, lua, bash, markdown, markdown_inline
-    '';
+let
+  inherit (lib) types;
+  inherit (lib.nixvim) defaultNullOpts;
+in
+lib.nixvim.neovim-plugin.mkNeovimPlugin {
+  name = "noice";
+  originalName = "noice.nvim";
+  package = "noice-nvim";
+
+  maintainers = [ lib.maintainers.khaneliman ];
+
+  description = ''
+    `noice.nvim`, an experimental Neovim UI.
+
+    > [!NOTE] If treesitter is enabled you need the following parsers:
+    >  vim, regex, lua, bash, markdown, markdown_inline
+  '';
 
-    package = lib.mkPackageOption pkgs "noice" {
-      default = [
-        "vimPlugins"
-        "noice-nvim"
+  # TODO: added 2024-10-27 remove after 24.11
+  deprecateExtraOptions = true;
+  optionsRenamedToSettings =
+    let
+      mkOptionPaths = paths: map (p: lib.splitString "." p) paths;
```

```suggestion
      mkOptionPaths = map (lib.splitString ".");
```

No need to specify the arg if it is passed as the final arg anyway. We can just use a partially applied function.

Same for the mapping function.